gcsio/src/test/java/com/google/cloud/hadoop/gcsio/EventLoggingHttpRequestInitializerTest.java

```diff
@@ -0,0 +1,111 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.LowLevelHttpRequest;
+import com.google.api.client.http.LowLevelHttpResponse;
+import com.google.api.client.testing.http.MockHttpTransport;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class EventLoggingHttpRequestInitializerTest {
+  private static final String requestUrl = "http://google.com";
+
+  @Mock private LowLevelHttpRequest mockLowLevelRequest;
+  @Mock private LowLevelHttpResponse mockLowLevelResponse;
+
+  private HttpRequest httpRequest;
+  private EventLoggingHttpRequestInitializer requestInitializer;
+  private EventLoggingHttpRequestInitializer requestInitializerSpy;
+  private ArgumentCaptor<HashMap<String, Object>> logDetailsCaptor;
```

Comment:
Maybe it is better to test the serialized json as argument, so we also test that logged format is indeed json.

Reply:
1. If we have to test this, then we will have to mock GoogleLogger static methods, which is not straightforward to do. Note that creating a method with string is not an option due to performance reasons (i.e. we want to pay the cost of creating the log only if the log is enabled).
2. The functionality of converting a map to Json is from GSON library and there is not much value testing that in this classes UT.